src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -40,18 +40,22 @@ public class MutableObjectModelBinder : IModelBinder
                 return null;
             }
 
-            EnsureModel(bindingContext);
-            var result = await CreateAndPopulateDto(bindingContext, mutableObjectBinderContext.PropertyMetadata);
+            // Create model first (if necessary) to avoid reporting errors about properties when activation fails.
+            var model = GetModel(bindingContext);
+
+            var results = await BindPropertiesAsync(bindingContext, mutableObjectBinderContext.PropertyMetadata);
```

Comment:
it would be good to do this property-by-property rather than "bind all properties" *then* "set all properties" - I don't think it needs to happen as part of this PR, but it should eventually for perf reasons.